03-components/30-removable-emails/EmailList.js

```diff
@@ -15,13 +15,16 @@ export default defineComponent({
     },
   },
 
+  emits: ['removeEmailByIndex'],
+
   template: `
     <ul class="emails-list unstyled-list" aria-label="Emails">
       <EmailListItem
         v-for="({ email, isMarked }, index) in emails"
         :key="email"
         :email="email"
         :marked="isMarked"
+        @removeEmailByIndex="$emit('removeEmailByIndex', index)"
```

Comment:
In templates style guide recommends `kebab-case` for events and props: `@remove-email-by-index`.
Also events are usually nouns, name of the action. `click`, `change`. "What happened? Click. Remove.". So, here a preferred name could be `remove`, `remove-email`.

Reply:
I thought it should be a verb. Thanks, I'll remember. Kebab-case - ok.